File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/242-ValidAnagram/242-ValidAnagram.java
// Last updated: 8/4/2026, 7:17:58 PM
class Solution {
    public boolean isAnagram(String s, String t) {
        int arr [] = new int [26];
        if(s.length() != t.length()) return false;

        for(int i = 0 ; i < s.length() ; i++){ 
            int a = s.charAt(i) - 'a';
            int b = t.charAt(i) - 'a';

            arr[a]++;
            arr[b]--;
        } 
        for(int i = 0 ; i < arr.length ; i++){
            if(arr[i] != 0) return false;
        }
        return true;
    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 8/4/2026, 7:17:52 PM
class Solution {
public:
    bool isAnagram(string s, string t) {
        int n = s.length();
        int m = t.length();

        if(n != m) return false;

        int arr[26];
        

        for(int i = 0 ; i < n ; i++){
            arr[s[i]-'a']++;
            arr[t[i]-'a']--;
        }

        for(int i = 0 ; i < 26 ; i++){
            if(arr[i] != 0){
                return false;
            }
        }
        return true;
        
    }
};